src/lib/FreeC/Backend/Agda/Converter/Free.hs

```diff
@@ -32,7 +33,22 @@ applyFreeArgs qname = foldl1
   , Agda.Ident (Agda.qname' Agda.Base.position)
   ]
 
--- | Adds the resvered names for the free args @Shape@ and @Pos@ to a list of
+-- | Adds the reserved names for the free args @Shape@ and @Pos@ to a list of
 --   names.
 addFreeArgs :: [Agda.Name] -> [Agda.Name]
 addFreeArgs ts = Agda.Base.shape : Agda.Base.position : ts
+
+-- | Identifier for @Shape@.
+--
+--   > Shape
+shape :: Agda.Expr
+shape = Agda.Ident $ Agda.qname' $ Agda.Base.shape
```

Comment:
If you cannot decide between explicit parentheses and `$`, please always go for parentheses instead! (In line 32 you use parentheses, for example.)

Reply:
I usually use `$` when applying two or more functions to the same argument to avoid writing nested parentheses. I think `f (g (h x))` is harder to read then `f $ g $ h x`. I can of course change it.